### ISSUE TITLE
Getting "Illegal Hardware Instruction" for python3.7 when using psychopy.hardware.keyboard

### ISSUE THREAD

**Issue by Catyre** [psychopy/psychopy#4342]
I am running psychopy 2021.2.3 on MacOS Big Sur 11.6.  Whenever I use psychopy.hardware.keyboard, I get an error in the terminal saying "illegal hardware instructions," followed shortly by a crash report.  This can be replicated in even the most sterile of conditions:

```
from psychopy.hardware import keyboard 
test = keyboard.Keyboard()
```

Full crash report can be found on pastebin: https://pastebin.com/1RxbHfnm

After some research into related problems (see [this](https://stackoverflow.com/questions/65383338/zsh-illegal-hardware-instruction-python-when-installing-tensorflow-on-macbook/68214296) StackOverflow question and the linked question at the top), it seems like there is a machine code command for one architecture (I suspect arm64) that is not recognized by my environment's architecture (x86_64).  Attempts to explicitly install an x86_64 psychopy (`arch -x86_64 $(which python3.7) -m pip install psychopy`) yielded nothing, although I haven't tried doing the same for its extensive list of dependencies for time reasons.

More information about my environment can be given upon request.  I'm just not sure what else is needed.

Any help would be appreciated.  Thanks in advance!

**Comment by bollaking** [comment]
I think this is the same as: 
https://github.com/psychopy/psychopy/issues/4329

There is also a bug reported in psychtoolbox to fix this for python 3.8. 
https://github.com/Psychtoolbox-3/Psychtoolbox-3/issues/744

**Comment by isolver** [comment]
Yes, for now please use python 3.6 on macOS if using ptb based keyboard.

**Comment by Catyre** [comment]
Okay, so I've installed python 3.6 and tried again, but I'm still getting the same error.

**Comment by bollaking** [comment]
I have been unable to install python 3.6 using homebrew or compiling from scratch.  It is no longer supported as of macos 11.0

I have been able to run experiments by raising an ImportError exception on line 73 of hardware.keyboard.py to force psychopy to use the events for keyboard events.

**Comment by Catyre** [comment]
Looks like there's already a codeblock in my version of psychopy for that:

```
try:
    import psychtoolbox as ptb
    from psychtoolbox import hid
    havePTB = True

except ImportError as err:
    logging.warning(("Import Error: "
                     + err.args[0]
                     + ". Using event module for keyboard component."))
    from psychopy import event
    havePTB = False
```

Were you saying that adding that fixed your (our) problem?

**Comment by bollaking** [comment]
Sorry I meant to raise the ImportError so that psychopy does not use ptb ie:
```
try:
    raise ImportError('Use event')
    import psychtoolbox as ptb
    from psychtoolbox import hid
    havePTB = True

except ImportError as err:
    logging.warning(("Import Error: "
                     + err.args[0]
                     + ". Using event module for keyboard component."))
    from psychopy import event
    havePTB = False
```

This is just a temporary hack to be able to run experiments in built locally until ptb can run with 3.8

**Comment by Catyre** [comment]
EDIT: Unrelated problem.  Removed it.

 My environment works now!  Thanks so much
